portfolio/src/main/webapp/script.js

```diff
@@ -38,18 +37,100 @@ function showModal(img) {
   console.log("Fetching data.");
-  const promise = fetch("/data").then(response => response.json()).then(greeting => {
-    console.log(greeting);
-    document.getElementById('data-container').innerText = greeting;
+  const maxElement = document.getElementById("quantity");
+  const maxToDisplay = maxElement.value;
+  const sort = document.getElementById("sort").value;
+  
+  const imgsrc = document.getElementById("modal-img").src;
+  const splitimg = imgsrc.split("/");
+  console.log(splitimg);
+  // Get image file name in img tag's src field (ex: flowers.jpg)
+  const splitname = splitimg[splitimg.length - 1].split(".");
+  // Get string before file format string (remove .jpg)
+  const img = splitname[0];
+
+  console.log(sort);
+  console.log(maxToDisplay);
+  
+  fetch("/comment?quantity=" + maxToDisplay + "&sort=" + sort + "&img=" + img).then(response => response.json()).then(jsonObj => {
+    const comments = jsonObj.comments;
+    console.log(comments[comments.length - 1]);
+    const total = jsonObj.total;
+    console.log(total);
+
+    const totalElement = document.getElementById("total");
+    totalElement.innerText = ("Comments (" + total + ")");
+
+    // Set maximum number of comments to show as the total number of comments (returned from fetch)
+    maxElement.max = total.toString();
+    maxElement.placeholder = total.toString();
+
+    // Add individual comments to html page
+    const list = document.getElementById("all-comments");
+    list.innerHTML = "";
+    comments.forEach((comment) => {
+      const listElement = document.createElement("li");
+
+      const textElement = document.createElement("span");
+      textElement.innerHTML = comment.name + "<br><br>" + comment.comment;
+
+      const deleteButtonElement = document.createElement("button");
+      deleteButtonElement.className = "delete-button";
+      deleteButtonElement.innerText = "Delete";
+      deleteButtonElement.addEventListener("click", () => {
+        console.log("Delete button clicked");
+        deleteComment(comment);
+      });
+      
+      listElement.appendChild(textElement);
+      listElement.appendChild(deleteButtonElement);
+      list.append(listElement);
+    })
   });
-}
\ No newline at end of file
+}
+
+function deleteComment(comment) {
+  console.log("Deleting comment");
+
+  const params = new URLSearchParams();
+  params.append("id", comment.id);
+  const postRequest = new Request("/delete", {method: "POST", body: params});
+  fetch(postRequest).then(() => getServerData()).catch(error => console.error(error));
+}
+
+function postComment() {
+  console.log("Posting comment");
+
+  const nameElement = document.getElementById("name");
+  const commentElement = document.getElementById("comment-box");
+  const imgsrc = document.getElementById("modal-img").src;
+  const splitimg = imgsrc.split("/");
+  console.log(splitimg);
+  const splitname = splitimg[splitimg.length - 1].split(".");
+  const img = splitname[0];
```

Comment:
May be worth to create a helper function?

Reply:
I agree!